hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/Sinks.java

```diff
@@ -716,19 +723,70 @@ private Sinks() {
 
     /**
      * Returns a builder object that offers a step-by-step fluent API to build
-     * a custom file sink for the Pipeline API. See javadoc of methods in {@link
-     * FileSinkBuilder} for more details.
-     * <p>
-     * The sink writes the items it receives to files. Each
-     * processor will write to its own file whose name is equal to the
-     * processor's global index (an integer unique to each processor of the
-     * vertex), but a single pathname is used to resolve the containing
-     * directory of all files, on all cluster members.
-     * <p>
-     * No state is saved to snapshot for this sink. If the job is restarted and
-     * {@linkplain FileSinkBuilder#append(boolean) appending} is enabled, the
-     * items will likely be duplicated, providing an <i>at-least-once</i>
-     * guarantee.
+     * a custom file sink for the Pipeline API. See javadoc of methods in
+     * {@link FileSinkBuilder} for more details.
+     * <p>
+     * The sink writes the items it receives to files. Each processor will
+     * write to its own files whose names contain the processor's global index
+     * (an integer unique to each processor of the vertex), but a single
+     * directory is used for all files, on all cluster members. That directory
+     * can be a shared in a network - the global processor index is different
+     * on each member and the members won't overwrite each other's files.
+     *
+     * <h3>Fault tolerance</h3>
+     * If the job is running in <i>exactly-once</i> mode, items will be written
+     * to temporary files (ending with a {@value #TEMP_FILE_SUFFIX} suffix).
+     * When the snapshot is committed, the file will be atomically renamed to
+     * remove this suffix. Thanks to the two-phase commit of the snapshot,
+     * exactly-once guarantee is provided for the sink. Because of this a new
+     * file will be started each time a state snapshotted, the sink will likely
+     * produce many more small files, depending on the snapshot interval.
+     * <p>
+     * If you want to avoid the temporary files or the high number of files but
+     * need to have exactly-once for other processors in the job, call {@link
+     * FileSinkBuilder#exactlyOnce(boolean) exactlyOnce(false)} on the returned
+     * builder. This will give you <i>at-least-once</i> guarantee for the
+     * source and unchanged guarantee for other processors.
+     *
+     * <h3>File name structure</h3>
+     * <pre>{@code
+     * [<date>-]<global processor index>[-<sequence>][".tmp"]
+     * }</pre>
+     *
+     * Legend (parts in {@code []} are optional):
```

Comment:
"Description" might be a better word for this, than "legend". Plus everybody knows that the [] are optional parts (or if they don't, they are too stupid to use Jet anyways).